src/services/Store/store/asset.slice.ts

```diff
@@ -67,3 +72,20 @@ export const {
 } = slice.actions;
 
 export default slice;
+
+/**
+ * Sagas
+ */
+export function* assetSaga() {
+  yield all([
+    takeLatest(fetchAssets.type, fetchAssetsWorker),
+    // Trigger fetching assets on RESET, is dispatched when resetting and importing new settings
+    takeLatest(ActionT.RESET, fetchAssetsWorker)
+  ]);
+}
+
+export function* fetchAssetsWorker() {
+  const fetchedAssets = yield call(MyCryptoApiService.instance.getAssets);
```

Comment:
Let's place our api calls inside a try/catch and add dispatch an error action to an Error reducer.
1. It anticipates centralised error handling.
2. It avoids errors bubbling to react.